omod/src/main/webapp/resources/scripts/conditionlist/controllers/addcondition.controller.js

```diff
@@ -21,20 +21,21 @@ function ConditionController($scope, RestfulService, ConditionModel, ConceptMode
      * @type {Function}
      */
     self.saveCondition = self.saveCondition || function () {
-        if($scope.condition.status == INACTIVE_STATUS && $scope.condition.onSetDate > $scope.condition.endDate){
+        if($scope.condition.status == INACTIVE_STATUS && $scope.condition.onSetDate > $scope.condition.endDate) {
             emr.errorAlert("End Date can't be before Onset Date");
         }
-        else{
+        else {
             var conditions = [];
             conditions.push($scope.condition);
-            RestfulService.post('condition', conditions, function (data) {
-                //emr.successAlert("conditionlist.updateCondition.success"); Messages not being resolved
-                emr.successAlert("Condition added successfully")
-                window.location = '/' + OPENMRS_CONTEXT_PATH + '/coreapps/conditionlist/manageConditions.page?patientId=' + $scope.patientUuid + '&';
-            }, function (error) {
-                //emr.errorAlert("conditionlist.updateCondition.error");
+            if (conditions[0].conditionNonCoded === "NON_CODED:" + $scope.concept.word) {
```

Comment:
We should save non-coded condition in the case when the `conditionList.nonCodedUuid` GP is set